src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/IApplicationPartTypeProvider.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// Gives access to a set of types from an <see cref="ApplicationPart"/>.
+    /// </summary>
+    public interface IApplicationPartTypeProvider
```

Comment:
Will #4344 be updated to be based off this PR? Hard to tell what's different in that PR. BTW if they are different branches I think you can send a PR that is just the diff from this PR (if you make it a sub-branch).